coolkit/dependencies/dependency.py

```diff
@@ -69,42 +73,47 @@ def set_global_config():
     shutil.copy(path_of_default_global_config, path)
 
-def install_dependencies(dependency_map, verbose = False):
-    supported_distros = _get_supported_distros(dependency_map)
-    distro = get_distro(supported_distros)
-    if(not distro and verbose):
-        print_err('unrecognised distro, please contact srbcheema2@gmail.com for full support for your distro',C.R)
-    elif(distro and verbose):
-        print_err('Distro detected to be '+distro+' based',C.G)
 
+def install_dependencies(dependency_map, verbose = False):
     all_installed = True
+    package_manager = get_available_package_manager(dependency_map)
+
+#--handling-absence-of-package-manager--
+    if not package_manager :
+        all_installed = False
+        if verbose :
+            print_err('Error: Unrecognised package manager',C.R)
+            print_err('Please contact srbcheema2@gmail.com for full support for your system.')
+
+            for dependency in dependency_map.keys() :
+                if not is_installed(dependency) :
+                    print_err('Please install ' + dependency + ' dependency manually',C.R)
+#--absence-of-package-manager-handled-
+
+    else :
+        if verbose :
+            print_clr('Package manager detected to be '+ C.E + package_manager , C.G) #distinction of package manager
+
+        for dependency in dependency_map.keys():
+
+            if is_installed(dependency):
+                if verbose :
+                    print_clr('.:Dependency '+dependency+' already installed...',C.G)
+                continue
+
+            rules = dependency_map[dependency]
+
+            print_clr('.:Installing ' + C.E + dependency + C.G +' dependency' , C.G) #distinction of dependency
+            os.system(rules[package_manager])
 
-    for d in dependency_map.keys():
-        if is_installed(d):
-            continue
-        rules = dependency_map[d]
-        if distro and distro in rules.keys():
-            print_err('installing '+d+' dependency',C.G)
-            os.system(rules[distro])
-            if not is_installed(d):
-                print_err('please install ' +d+ ' dependency manually',C.Y)
-                print_err('try command : '+rules[distro],C.Y)
+            if not is_installed(dependency):
+                print_err('please install ' + dependency + ' dependency manually',C.Y)
+                print_err('try command : '+rules[package_manager],C.Y)
                 all_installed = False
-        else:
-            print_err('Please install ' +d+ ' dependency manually',C.R)
-            all_installed = False
 
     return all_installed
 
 
 if __name__ == '__main__':
-    dependency_map = {
-        'register-python-argcomplete':{
-            'ubuntu':'sudo apt install python-argcomplete',
-        },
-        'figlet':{
-            'ubuntu':'sudo apt install figlet',
-        },
-    }
-    install_dependencies(dependency_map,verbose=True)
+    install_dependencies(dependencies.dependency_map,verbose=True)
```

Comment:
the code in main is just kept to test this script so I have used simple things to test. revert it as it is. this script is not to be run as main. main is just to test.